### ISSUE TITLE
new grid behaves strangely with corrplot: only appears where there are ticks

### ISSUE THREAD

**Issue by piever** [JuliaPlots/StatsPlots.jl#78]
`corrplot` recipe has a strange grid:

```julia
M = randn(1000,4)
M[:,2] += 0.8sqrt.(abs.(M[:,1])) - 0.5M[:,3] + 5
M[:,3] -= 0.7M[:,1].^2 + 2
corrplot(M, label = ["x$i" for i=1:4])
```

![gridcorrplot](https://user-images.githubusercontent.com/6333339/29819065-e7f5afb4-8cb6-11e7-9620-9b3290ca6d8f.png)

**Comment by mkborregaard** [comment]
Going back, it's been like this for a long time. I think we might just turn off the grid by default in corrplot :-D

**Comment by piever** [comment]
I actually never noticed because I have `grid = false` in my defaults... I asked because I was redoing the README for the new syntax and wanted to add a `corrplot` of `iris`. I'll simply do it without grid then.

**Comment by daschw** [comment]
It would be easy to fix e.g. by replacing `d[:xtick] = (i==n)` with `d[:xformatter] = i == n ? :auto : x -> ""` [here](https://github.com/JuliaPlots/StatPlots.jl/blob/master/src/corrplot.jl#L5) and for analogously for `y` in L7.
I'm not sure if it makes sense, though, because the `yticks` of the upper left subplot don't correspond to the `yticks` of the other subplots in that row and the `xticks` of the lower right subplot don't correspond to the `xticks` of the other subplots in that column. So it would be more "correct" to either show the ticks in all subplots or in none, IMHO.

**Comment by mkborregaard** [comment]
ah, because of the histograms. I see. Hmm then perhaps show in all?

**Comment by mkborregaard** [comment]
On the other hand, that will be extremely busy (and probably unneccesary) on larger corrplots.

**Comment by daschw** [comment]
Well, it can always be turned off easily by the user with `ticks = false`, right?

**Comment by mkborregaard** [comment]
Sure, just trying to find the best default. I think in fact this might be a nice candidate for your framestyle = :grid?

**Comment by daschw** [comment]
The ticks would be still there though, but maybe it looks less crowded without the axes.

**Comment by daschw** [comment]
with ticks
![corrplot_ticks](https://user-images.githubusercontent.com/16589944/29843044-ff66224c-8d0a-11e7-8efc-27ba2811f6bf.png)

`framestyle = :grid`
![corrplot_grid](https://user-images.githubusercontent.com/16589944/29843070-10d3eb90-8d0b-11e7-8f1f-72ac937f58df.png)

`ticks = false`
![corrplot_noticks](https://user-images.githubusercontent.com/16589944/29843052-05901ce0-8d0b-11e7-9620-d3ee27083f07.png)

**Comment by daschw** [comment]
The last one looks definitely cleanest.

**Comment by mkborregaard** [comment]
Yes, and it brings out what we are most interested in with the corrplot

**Comment by piever** [comment]
Closed by #81

**Comment by mauro3** [comment]
So how can I get the corrplot back with tick-labels just along the outermost plots?  `ticks=true` puts them everywhere: https://github.com/JuliaPlots/StatPlots.jl/issues/78#issuecomment-325797317

Also, is the disappearance of tick-labels in the cornerplot along the bottom related to this?
